Gem/Code/Source/Components/NetworkTeleportComponent.cpp

```diff
@@ -0,0 +1,141 @@
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, 
+                        "MultiplayerSample")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, 
+                        AZ_CRC_CE("Game"))
+                    ->DataElement(0, &NetworkTeleportComponent::m_reset,
+                        "Destination", "Entity at which to teleport the colliding player")
+                    ;
+            }
+        }
+    }
+
+    NetworkTeleportComponent::NetworkTeleportComponent()
+        : m_enterTrigger([this](
+            AzPhysics::SimulatedBodyHandle bodyHandle, 
+            const AzPhysics::TriggerEvent& triggerEvent)
+            {
+                this->OnTriggerEnter(bodyHandle, triggerEvent);
+            })
+    {
+    }
+
+    void NetworkTeleportComponent::Activate()
+    {
+        auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
+         if (physicsSystem)
+        {
+            auto [sceneHandle, bodyHandle] = physicsSystem->FindAttachedBodyHandleFromEntityId(GetEntityId());
+            AzPhysics::SimulatedBodyEvents::RegisterOnTriggerEnterHandler(
+                sceneHandle, bodyHandle, m_enterTrigger);
+        }
+    }
+
+    void NetworkTeleportComponent::OnTriggerEnter(
+        [[maybe_unused]] AzPhysics::SimulatedBodyHandle bodyHandle, const AzPhysics::TriggerEvent& triggerEvent)
+    {
+
+        if (triggerEvent.m_otherBody)
+        {
+            // TODO: validate it's a player entity
+            AZ::Vector3 teleportLocation = GetDestination();
+            AZ_TracePrintf("Teleporter", "destination point X is: %f\n",
+                teleportLocation.GetX());
+
+            AZ::Entity* otherEntity = GetCollidingEntity(triggerEvent.m_otherBody);
+            TeleportPlayer(teleportLocation, otherEntity);
+        }
+    }
+
+    AZ::Entity* NetworkTeleportComponent::GetCollidingEntity(AzPhysics::SimulatedBody* collidingBody) const
+    {
+        AZ::Entity* collidingEntity = nullptr;		
+        if (collidingBody)
+        {
+            AZ::EntityId collidingEntityId = collidingBody->GetEntityId();
+            AZ::ComponentApplicationBus::BroadcastResult(
+                collidingEntity, &AZ::ComponentApplicationBus::Events::FindEntity, collidingEntityId);
+        }
+        return collidingEntity;
+    }
+
+    AZ::Vector3 NetworkTeleportComponent::GetDestination() const
+    {
+        AZ::Vector3 newLocation = AZ::Vector3::CreateZero();
+        AZ::TransformBus::EventResult(newLocation, m_reset,
+            &AZ::TransformBus::Events::GetWorldTranslation);
+
+        return newLocation;
+    }
+
+    void NetworkTeleportComponent::TeleportPlayer(
+        const AZ::Vector3& v, AZ::Entity* entity)
+    {
+        if (entity)
+        {
+            AZ::EntityId playerToTeleport = entity->GetId();
+
+            AZ_Printf(
```

Comment:
I imagine this can be removed?

Reply:
I though it would be useful to visibly log one message for the event, and this seemed the like the most useful info. But maybe even that would cause too much spam given enough events?

Reply:
If you think it's useful long term, I suggest changing it to AZ_TracePrintf.